doc/releases/changelog-dev.md

```diff
@@ -120,6 +120,10 @@
 
 <h3>Deprecations 👋</h3>
 
+* Legacy operator arithmetic has been deprecated. This includes `qml.ops.Hamiltonian`, `qml.operation.enable_new_opmath`,
+  `qml.operation.disable_new_opmath`, and `qml.operation.convert_to_legacy_H`.
+  [(#6287)](https://github.com/PennyLaneAI/pennylane/pull/6287)
+
```

Comment:
I am not suggesting doing that in this PR, but are we going to remove the `Tensor` class without deprecating it first? If so, then I guess that this can cause quite a few issues

Reply:
I believe the logic behind not deprecating `Tensor` is that it is a private class. We've always exposed `qml.Hamiltonian` in a very public way, but `Tensor` has always been hidden inside the `operation` module, and the recommended way to create a `Tensor` has been using the `@` operator. That being said, I see your point, and I'd feel wrong merging this PR before getting more input from product about it. cc @DSGuala

Reply:
After more discussions, `Tensor` has been deprecated.